### PR TITLE
WIP authors: extend publications with affiliations

### PR DESCRIPTION
**Pull request by jacenkow**
- Extends /authors API with affiliation field for every
  publication. Referes #1090.
- Includes affiliation in the data table of author profile.

Signed-off-by: Grzegorz Jacenków grzegorz.jacenkow@cern.ch

### CODE REVIEW

**Comment by jacenkow** [comment]
CC: @kaplun, @jacquerie

File: tests/integration/test_author_api.py [hunk 1]
```diff
@@ -60,7 +60,9 @@ def test_api_authors_citations(app):
                                 'properties': {
                                     'id': {'type': 'integer'},
                                     'record': {
-                                        'properties': {'$ref': {'type': 'string'}},
+                                        'properties': {'$ref': {
+                                            'type': 'string'}
+                                        },
```
Comment (jacquerie) [inline tests/integration/test_author_api.py]: This guy can stay like it was, no?

Comment (jacenkow) [inline tests/integration/test_author_api.py]: only because of 83 chars 😛

File: inspirehep/modules/authors/rest/publications.py [hunk 2]
```diff
@@ -73,6 +75,25 @@ def serialize(self, pid, record, links_factory=None):
...
+            # Get author's affiliation.
+            for author in result_source['authors']:
+                try:
+                    if author['recid'] == int(author_pid):
+                        publication['affiliation'] = {}
+                        try:
+                            publication['affiliation']['name'] = author[
+                                'affiliations'][0]['value']
+                        except (IndexError, KeyError):
+                            del publication['affiliation']
```
Comment (jacquerie) [inline inspirehep/modules/authors/rest/publications.py]: Can you explain the logic here?

File: tests/integration/test_author_api.py (hunk 1 above)
Comment (jacquerie) [inline tests/integration/test_author_api.py]: That's why we ignore E501!

**Comment by jacquerie** [comment]
My problem with merging this is that I cannot run it locally due to some problem in the Angular.js injector, even when `DEBUG = True`

CC: @jmartinm

**Comment by kaplun** [comment]
@jmartinm as mentioned today you can take it over.

**Comment by jacenkow** [comment]
Nice 🎱

**Comment by jacquerie** [comment]
Postponed, it will be resurrected if necessary.

**Pull request closed without merging**